src/transport/NetworkProvisioning.cpp

```diff
@@ -161,20 +166,17 @@ CHIP_ERROR NetworkProvisioning::DecodeString(const uint8_t * input, size_t input
 
 CHIP_ERROR NetworkProvisioning::SendIPAddress(const Inet::IPAddress & addr)
 {
+    char * addrStr;
     CHIP_ERROR err                    = CHIP_NO_ERROR;
-    System::PacketBufferHandle buffer = System::PacketBuffer::New();
-    char * addrStr                    = addr.ToString(Uint8::to_char(buffer->Start()), buffer->AvailableDataLength());
-    size_t addrLen                    = 0;
+    System::PacketBufferHandle buffer = System::PacketBufferHandle::New(Inet::kMaxIPAddressStringLength);
+    VerifyOrExit(!buffer.IsNull(), err = CHIP_ERROR_NO_MEMORY);
+    addrStr = addr.ToString(Uint8::to_char(buffer->Start()), buffer->AvailableDataLength());
+    buffer->SetDataLength(static_cast<uint16_t>(strlen(addrStr) + 1));
```

Comment:
Why the +1? Should this be +2 ?

Reply:
It matches what the code _did_, but maybe that's wrong. Is it supposed to use `EncodeString`?